### PR TITLE
[Enhancement] Make routine load error msg more clear

### PR DESCRIPTION
**Pull request by wyb**
## What I'm doing:

1. Improve kafka error message
2. Set the timeout of BE get kafka topic partition info to 80% of the rpc timeout
3. The task error message is shown in show routine load
4. Refactor `KafkaUtil.sendProxyRequest`
5. Fix `consumer.metadata` function `all_topics` parameter

error message:
1. Kafka broker is down
```
mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 106275
                Name: label0
          CreateTime: 2024-02-21 15:52:17
           PauseTime: 2024-02-21 17:17:54
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: PAUSED
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"0","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"0","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_1f9af280-5fef-44d1-85bf-44f41c4c7fcd"}
           Statistic: {"receivedBytes":2281306,"errorRows":0,"committedTaskNum":1,"loadedRows":22683,"loadRowsRate":1000,"abortedTaskNum":0,"totalRows":22683,"unselectedRows":0,"receivedBytesRate":151000,"taskExecuteTimeMs":15040}
            Progress: {"0":"501572079"}
   TimestampProgress: {}
ReasonOfStateChanged: ErrorReason{errCode = 2, msg='failed to get kafka topic: wyb_test meta, err: Local: Broker transport failure, event: [thrd:1.1.1.1:9093/bootstrap]: 1.1.1.1:9093/bootstrap: Connect to ipv4#1.1.1.1:9093 failed: Connection refused (after 0ms in state CONNECT), BE: TNetworkAddress(hostname:1.1.1.1, port:8069)'}
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg:
LatestSourcePosition: {}
1 row in set (0.01 sec)

mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 106275
                Name: label0
          CreateTime: 2024-02-21 15:52:17
           PauseTime: 2024-02-21 17:17:54
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: RUNNING
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"0","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"0","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_1f9af280-5fef-44d1-85bf-44f41c4c7fcd"}
           Statistic: {"receivedBytes":2281306,"errorRows":0,"committedTaskNum":1,"loadedRows":22683,"loadRowsRate":1000,"abortedTaskNum":0,"totalRows":22683,"unselectedRows":0,"receivedBytesRate":151000,"taskExecuteTimeMs":15040}
            Progress: {"0":"501572079"}
   TimestampProgress: {}
ReasonOfStateChanged:
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg: [2024-03-19 18:39:29] [task id: 12178177-28e4-49e7-93e3-af5c0a06bbcb] [txn id: -1] check task ready to execute failed, err: failed to get offset of partition: 0 in topic: wyb_test, err: Local: All broker connections are down, event: [thrd:1.1.1.1:9093/bootstrap]: 1.1.1.1:9093/bootstrap: Connect to ipv4#1.1.1.1:9093 failed: Connection refused (after 0ms in state CONNECT), BE: TNetworkAddress(hostname:1.1.1.1, port:8069)
LatestSourcePosition: {}
1 row in set (0.01 sec)
```

2. BE is down
```
mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 128134
                Name: label0
          CreateTime: 2024-03-07 15:42:21
           PauseTime: NULL
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: RUNNING
      DataSourceType: KAFKA
      CurrentTaskNum: 1
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"1","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"0","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_d704e31e-a674-4a76-81c2-113b395ff1fb"}
           Statistic: {"receivedBytes":90229443,"errorRows":0,"committedTaskNum":10,"loadedRows":906093,"loadRowsRate":5000,"abortedTaskNum":4,"totalRows":906093,"unselectedRows":0,"receivedBytesRate":587000,"taskExecuteTimeMs":153508}
            Progress: {"0":"2164591389"}
   TimestampProgress: {"0":"1709797992856"}
ReasonOfStateChanged:
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg: [2024-03-19 18:28:37] [task id: null] [txn id: -1] check task ready to execute failed, err: Failed to send get kafka partition info request to BE TNetworkAddress(hostname:1.1.1.1, port:8069). err: BE is not alive
LatestSourcePosition: {"0":"2164333309"}
1 row in set (0.00 sec)

mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 127092
                Name: label0
          CreateTime: 2024-03-07 14:28:39
           PauseTime: NULL
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: RUNNING
      DataSourceType: KAFKA
      CurrentTaskNum: 1
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"1","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"0","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_37e82761-e45e-4686-a352-7b158cfa6c37"}
           Statistic: {"receivedBytes":196467825,"errorRows":0,"committedTaskNum":18,"loadedRows":1972935,"loadRowsRate":7000,"abortedTaskNum":4,"totalRows":1972935,"unselectedRows":0,"receivedBytesRate":746000,"taskExecuteTimeMs":263315}
            Progress: {"0":"2161445207"}
   TimestampProgress: {"0":"1709795800554"}
ReasonOfStateChanged:
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg: [2024-03-19 18:28:37] [task id: null] [txn id: -1] check task ready to execute failed, err: Failed to send get kafka partition info request. err: No alive backends
LatestSourcePosition: {"0":"2161425566"}
1 row in set (0.00 sec)
```

3. Unknown kafka broker host
```
mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 107096
                Name: label0
          CreateTime: 2024-02-21 17:34:24
           PauseTime: 2024-02-21 17:35:04
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: PAUSED
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"0","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"0","brokerList":"test_host:9093"}
    CustomProperties: {"group.id":"label0_18adab03-c776-472a-80ba-41ecca292f98"}
           Statistic: {"receivedBytes":0,"errorRows":0,"committedTaskNum":0,"loadedRows":0,"loadRowsRate":0,"abortedTaskNum":0,"totalRows":0,"unselectedRows":0,"receivedBytesRate":0,"taskExecuteTimeMs":1}
            Progress: {"0":"OFFSET_END"}
   TimestampProgress: {}
ReasonOfStateChanged: ErrorReason{errCode = 2, msg='failed to get kafka topic: wyb_test meta, err: Local: Broker transport failure, event: [thrd:test_host:9093/bootstrap]: test_host:9093/bootstrap: Failed to resolve 'test_host:9093': Name or service not known (after 104ms in state CONNECT), BE: TNetworkAddress(hostname:xxx, port:8069)'}
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg:
LatestSourcePosition: {}
1 row in set (0.00 sec)
```

4. kafka topic or partition wrong
```
*************************** 1. row ***************************
                  Id: 130185
                Name: label0
          CreateTime: 2024-03-11 19:24:40
           PauseTime: 2024-03-11 20:47:37
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: PAUSED
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"0","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test1","currentKafkaPartitions":"0","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_6d2c08f5-1811-497d-a9ab-c210778b116f"}
           Statistic: {"receivedBytes":0,"errorRows":0,"committedTaskNum":0,"loadedRows":0,"loadRowsRate":0,"abortedTaskNum":0,"totalRows":0,"unselectedRows":0,"receivedBytesRate":0,"taskExecuteTimeMs":1}
            Progress: {"0":"OFFSET_END"}
   TimestampProgress: {}
ReasonOfStateChanged: ErrorReason{errCode = 2, msg='unknown topic: wyb_test1, BE: TNetworkAddress(hostname:1.1.1.1, port:8069)'}
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg:
LatestSourcePosition: {}
1 row in set (0.00 sec)

mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 129583
                Name: label0
          CreateTime: 2024-03-11 17:29:41
           PauseTime: 2024-03-11 17:47:13
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: PAUSED
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"0","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"0","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_049f54bc-e382-451f-ac8a-52acad998580"}
           Statistic: {"receivedBytes":0,"errorRows":0,"committedTaskNum":0,"loadedRows":0,"loadRowsRate":0,"abortedTaskNum":0,"totalRows":0,"unselectedRows":0,"receivedBytesRate":0,"taskExecuteTimeMs":1}
            Progress: {"0":"OFFSET_END"}
   TimestampProgress: {}
ReasonOfStateChanged: ErrorReason{errCode = 2, msg='there is no partition in topic: wyb_test, BE: TNetworkAddress(hostname:1.1.1.1, port:8069)'}
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg:
LatestSourcePosition: {}
1 row in set (0.00 sec)

mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 130096
                Name: label0
          CreateTime: 2024-03-11 17:52:00
           PauseTime: 2024-03-11 17:59:23
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: PAUSED
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"1","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"1","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_54457248-6b16-418b-8a55-6ff169469ab5"}
           Statistic: {"receivedBytes":0,"errorRows":0,"committedTaskNum":0,"loadedRows":0,"loadRowsRate":0,"abortedTaskNum":0,"totalRows":0,"unselectedRows":0,"receivedBytesRate":0,"taskExecuteTimeMs":1}
            Progress: {"1":"OFFSET_END"}
   TimestampProgress: {}
ReasonOfStateChanged: ErrorReason{errCode = 2, msg='there is an invalid custom partition: 1 in topic: wyb_test'}
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg:
LatestSourcePosition: {}
1 row in set (0.01 sec)
```

6. kafka partition offset has no data
```
mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 130163
                Name: label0
          CreateTime: 2024-03-11 19:02:10
           PauseTime: 2024-03-11 19:02:23
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: PAUSED
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"1","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"0","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_09eeebfb-c0f2-4a95-a684-a8aa43d38f06"}
           Statistic: {"receivedBytes":0,"errorRows":0,"committedTaskNum":0,"loadedRows":0,"loadRowsRate":0,"abortedTaskNum":0,"totalRows":0,"unselectedRows":0,"receivedBytesRate":0,"taskExecuteTimeMs":1}
            Progress: {"0":"2999999999"}
   TimestampProgress: {}
ReasonOfStateChanged: ErrorReason{errCode = 104, msg='fe abort task with reason: check task ready to execute failed, there is no data in partition: 0 at offset: 2999999999. you can modify kafka_offsets by alter routine load, then resume the job. refer to https://docs.starrocks.io/docs/sql-reference/sql-statements/data-manipulation/ALTER_ROUTINE_LOAD'}
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg:
LatestSourcePosition: {"0":"2661337494"}
1 row in set (0.00 sec)

mysql> show routine load\G
*************************** 1. row ***************************
                  Id: 130169
                Name: label0
          CreateTime: 2024-03-11 19:06:15
           PauseTime: NULL
             EndTime: NULL
              DbName: kafka_test
           TableName: lineorder
               State: RUNNING
      DataSourceType: KAFKA
      CurrentTaskNum: 1
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"*","maxBatchIntervalS":"10","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'|'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"5","maxErrorNum":"0","strip_outer_array":"false","currentTaskConcurrentNum":"1","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"wyb_test","currentKafkaPartitions":"0","brokerList":"1.1.1.1:9093"}
    CustomProperties: {"group.id":"label0_8dd51864-261a-4210-82f4-aa9829af4e04"}
           Statistic: {"receivedBytes":0,"errorRows":0,"committedTaskNum":0,"loadedRows":0,"loadRowsRate":0,"abortedTaskNum":0,"totalRows":0,"unselectedRows":0,"receivedBytesRate":0,"taskExecuteTimeMs":1}
            Progress: {"0":"2661337493"}
   TimestampProgress: {}
ReasonOfStateChanged:
        ErrorLogUrls:
         TrackingSQL:
            OtherMsg: [2024-03-19 18:42:16] [task id: c191344f-f78e-46bf-aec8-2b3d8abf0351] [txn id: -1] there is no new data in kafka, wait for 10 seconds to schedule again
LatestSourcePosition: {"0":"2661337494"}
1 row in set (0.00 sec)
```

7. wrong security protocol
```
mysql> show routine load \G
*************************** 1. row ***************************
                  Id: 10518
                Name: primary_routine_load_be4ddad6_e841_11ee_8ee4_00163e21975a
          CreateTime: 2024-03-22 20:17:11
           PauseTime: 2024-03-22 20:17:25
             EndTime: NULL
              DbName: test_db
           TableName: test_all_null1
               State: PAUSED
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"partitions":"*","rowDelimiter":"\t","partial_update":"false","columnToColumnExpr":"id_int,id_boolean,id_tinyint,id_smallint,id_bigint,id_largeint,id_float,id_double,id_decimal32,id_decimal64,id_decimal128,id_date,id_datetime,id_varchar","maxBatchIntervalS":"5","partial_update_mode":"null","whereExpr":"*","timezone":"Asia/Shanghai","format":"csv","columnSeparator":"'\t'","log_rejected_record_num":"0","taskTimeoutSecond":"60","json_root":"","maxFilterRatio":"1.0","strict_mode":"false","jsonpaths":"","taskConsumeSecond":"15","desireTaskConcurrentNum":"1","maxErrorNum":"1000","strip_outer_array":"false","currentTaskConcurrentNum":"1","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"condition_null_update","currentKafkaPartitions":"0,1,2,3,4","brokerList":"1.1.1.1:9092"}
    CustomProperties: {"security.protocol":"SSL","group.id":"primary_routine_load_be4ddad6_e841_11ee_8ee4_00163e21975a_8adbb113-96c4-46e8-b23d-0da9c897faaf"}
           Statistic: {"receivedBytes":0,"errorRows":0,"committedTaskNum":0,"loadedRows":0,"loadRowsRate":0,"abortedTaskNum":0,"totalRows":0,"unselectedRows":0,"receivedBytesRate":0,"taskExecuteTimeMs":1}
            Progress: {"0":"OFFSET_BEGINNING","1":"OFFSET_BEGINNING","2":"OFFSET_BEGINNING","3":"OFFSET_BEGINNING","4":"OFFSET_BEGINNING"}
   TimestampProgress: {}
ReasonOfStateChanged: ErrorReason{errCode = 2, msg='failed to get kafka topic: condition_null_update meta, err: Local: Broker transport failure, event: [thrd:ssl://1.1.1.1:9092/bootstrap]: ssl://1.1.1.1:9092/bootstrap: SSL handshake failed: Disconnected: connecting to a PLAINTEXT broker listener? (after 26ms in state SSL_HANDSHAKE), BE: TNetworkAddress(hostname:1.1.1.1, port:8060)'}
        ErrorLogUrls:
            OtherMsg:
LatestSourcePosition: {}
1 row in set (0.01 sec)
```

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by jaogoy** [comment]
All the errors occured when "to get kafka partition info"? How about in other scenarios?
In my opinion, the error info is mostly displayed, but the sentence is real long and redudant, it's not so clear to let users catch the real error quickly.

1. for "kafka broker is down"
    * Can you directly show the error message "Connection refused" in `err:[ ... ]` block without "failed to get kafka partition meta"? 
    * Can you directly put the error message from task into job's error message field? (The task's error message is clearer than the job's)
    * What's the use of "after 0ms in state CONNECT", if it's useless, can you just remove it.
    
2. for "BE is down"
    * What's the error message in the Job?
    * What's the meaning of "Unable to validate object"? It chould be "Failed to connec to BE: xxx".
    * The same about "Failed to send get kafka partition info request"?

**Comment by wyb** [comment]
> All the errors occured when "to get kafka partition info"? How about in other scenarios? In my opinion, the error info is mostly displayed, but the sentence is real long and redudant, it's not so clear to let users catch the real error quickly.
> 
> 1. for "kafka broker is down"
>    
>    * Can you directly show the error message "Connection refused" in `err:[ ... ]` block without "failed to get kafka partition meta"?
ok, directly return BE error message

>    * Can you directly put the error message from task into job's error message field? (The task's error message is clearer than the job's)
done

>    * What's the use of "after 0ms in state CONNECT", if it's useless, can you just remove it.
this is from kafka error message

> 2. for "BE is down"
>    
>    * What's the error message in the Job?
done

>    * What's the meaning of "Unable to validate object"? It chould be "Failed to connec to BE: xxx".
done

>    * The same about "Failed to send get kafka partition info request"?
better to keep this to let user know where is failed

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 1]
```diff
@@ -239,9 +241,11 @@ Status KafkaDataConsumer::group_consume(TimedBlockingQueue<RdKafka::Message*>* q
         case RdKafka::ERR_OFFSET_OUT_OF_RANGE: {
             done = true;
             std::stringstream ss;
-            ss << msg->errstr() << ", partition " << msg->partition() << " offset " << msg->offset() << " has no data";
+            ss << msg->errstr() << ", this is no data in partition: " << msg->partition()
```
Comment (rickif) [inline be/src/runtime/routine_load/data_consumer.cpp:244]: How about changing the error message to `offset out of range`, which is more common?

**Comment by rickif** [comment]
What will happen if the given kafka broker address is unreachable?

File: be/src/runtime/routine_load/data_consumer.cpp (hunk 1 above)
Comment (wyb) [reply be/src/runtime/routine_load/data_consumer.cpp:244]: users need to know kafka to understand `offset out of range`, this is specially modified before.

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 2]
```diff
@@ -351,9 +357,12 @@ Status KafkaDataConsumer::get_partition_meta(std::vector<int32_t>* partition_ids
             continue;
         }
 
-        if ((*it)->err() != RdKafka::ERR_NO_ERROR) {
+        if ((*it)->err() == RdKafka::ERR_UNKNOWN_TOPIC_OR_PART) {
+            LOG(WARNING) << "unknown topic: " << _topic;
```
[the remark this answers is not in the document]
Comment (wyb) [reply be/src/runtime/routine_load/data_consumer.cpp:361]: only topic metadata is processed here, so better to specify topic error clearly.

**Comment by wyb** [comment]
> What will happen if the given kafka broker address is unreachable, like `192.168.0.123` ?

this is `Kafka broker is down` case

**Review by rickif — APPROVED** [review]

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 3]
```diff
@@ -239,9 +241,11 @@ Status KafkaDataConsumer::group_consume(TimedBlockingQueue<RdKafka::Message*>* q
         case RdKafka::ERR_OFFSET_OUT_OF_RANGE: {
             done = true;
             std::stringstream ss;
-            ss << msg->errstr() << ", partition " << msg->partition() << " offset " << msg->offset() << " has no data";
+            ss << msg->errstr() << ", this is no data in partition: " << msg->partition()
+               << " at offset: " << msg->offset();
             LOG(WARNING) << "kafka consume failed: " << _id << ", msg: " << ss.str();
-            st = Status::InternalError(ss.str());
+            st = Status::InternalError(fmt::format(
+                    "{}. you can modify kafka_offsets by alter routine load, then resume the job", ss.str()));
```
Comment (banmoy) [inline be/src/runtime/routine_load/data_consumer.cpp:248]: How about giving a document link for how to alter routine load?

File: be/src/runtime/routine_load/data_consumer.cpp [hunk 4]
```diff
@@ -334,10 +339,11 @@ Status KafkaDataConsumer::get_partition_meta(std::vector<int32_t>* partition_ids
 
     // get topic metadata
     RdKafka::Metadata* metadata = nullptr;
-    RdKafka::ErrorCode err = _k_consumer->metadata(true /* for this topic */, topic, &metadata, timeout);
+    RdKafka::ErrorCode err = _k_consumer->metadata(false /* all_topics */, topic, &metadata, timeout);
```
Comment (banmoy) [inline be/src/runtime/routine_load/data_consumer.cpp:342]: As the api said, `false` only request info about locally known topics, not all topics ? 
```
/**
   * @brief Request Metadata from broker.
   *
   * Parameters:
   *  \p all_topics  - if non-zero: request info about all topics in cluster,
   *                   if zero: only request info about locally known topics.
   *  \p only_rkt    - only request info about this topic
   *  \p metadatap   - pointer to hold metadata result.
   *                   The \p *metadatap pointer must be released with \c delete.
   *  \p timeout_ms  - maximum response time before failing.
   *
   * @returns RdKafka::ERR_NO_ERROR on success (in which case \p *metadatap
   * will be set), else RdKafka::ERR__TIMED_OUT on timeout or
   * other error code on error.
   */
  virtual ErrorCode metadata (bool all_topics, const Topic *only_rkt,
                              Metadata **metadatap, int timeout_ms) = 0;
```

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadTaskInfo.java [hunk 5]
```diff
@@ -184,6 +184,8 @@ public long getTimeToExecuteMs() {
 
     public void setMsg(String msg) {
         this.msg = msg;
+        RoutineLoadJob routineLoadJob = routineLoadManager.getJob(jobId);
+        routineLoadJob.setOtherMsg(msg);
```
Comment (banmoy) [inline fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadTaskInfo.java:188]: Better to add `txnId`, `label` information to know which task the msg is from?

File: be/src/runtime/routine_load/data_consumer.cpp (hunk 3 above)
Comment (wyb) [reply be/src/runtime/routine_load/data_consumer.cpp]: done

File: fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadTaskInfo.java (hunk 5 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadTaskInfo.java]: done. 
txn id and task id are added.

File: be/src/runtime/routine_load/data_consumer.cpp (hunk 4 above)
Comment (wyb) [reply be/src/runtime/routine_load/data_consumer.cpp:345]: only the specified `topic` needs to get metadata

**Review by banmoy — APPROVED** [review]

**Review by meegoo — DISMISSED** [review]

**Review by meegoo — APPROVED** [review]

**Review by jaogoy — APPROVED** [review]

**Pull request merged**